router/config.go

```diff
@@ -885,12 +885,14 @@ func LoadConfigWithOptions(path string, loadIdentity bool) (*Config, error) {
 		for _, c := range cfg.Link.Listeners {
 			a := c["advertise"]
 			if a != nil {
-				// should start with tls:
-				addy := strings.TrimPrefix(a.(string), "tls:")
-				addy = strings.Split(addy, ":")[0]
-				e := cfg.Id.ValidFor(addy)
-				if e != nil {
-					errs = append(errs, fmt.Errorf("invalid link.listeners.advertise: %s, error: %v", a.(string), e))
+				// should start with tls: or dtls:
+				parts := strings.Split(a.(string), ":")
+				if parts[0] == "tls" || parts[0] == "dtls" {
```

Comment:
Seems odd that we would be explicitly checking for `tsl` and `dtls`. Are there other types of listeners we would be filtering out? The only one I can think of is the web socket listener on routers, but IIRC that isn't applicable for terminators.

Reply:
Clint had changed this to only allow tls, I'm broadening it to dtls. I think we can tweak it if/when we add another protocol. Might be able to check it against transport.ParseAddress.